docker/package/package_generator.py

```diff
@@ -66,113 +98,80 @@ def get_build_deps(args):
-
-ubuntu_versions = [
-    "bionic",  # 18.04
-    "focal",  # 20.04
-    "jammy",  # 22.04
-]
-
-pwd = os.getcwd()
-home = os.environ["HOME"]
-
-packages_to_build = packages
-
-if args.build_sapling_package:
-    packages.append(sapling_package)
-
-for package in packages_to_build:
-    if package_to_build is None or package.name == package_to_build:
-        if is_ubuntu:
+    args = parser.parse_args()
+
+    is_ubuntu = args.os == "ubuntu"
+
+    is_source = args.type == "source"
+
+    source_archive = args.sources
+
+    binaries_dir = args.binaries_dir
+
+    if is_ubuntu:
+        run_deps = get_ubuntu_run_deps(binaries_dir)
+    else:
+        run_deps = get_fedora_run_deps(binaries_dir)
+
+    build_deps = get_build_deps(binaries_dir)
+
+    common_deps = run_deps + build_deps
+
+    home = os.environ["HOME"]
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    distributions = list(args.distributions)
+
+    packages = []
+
+    for package_name in args.packages:
+        packages.append(all_packages[package_name])
+
+    if is_ubuntu:
+        for package in packages:
             build_ubuntu_package(
                 package,
-                ubuntu_versions,
+                distributions,
                 common_deps,
                 is_source,
                 source_archive,
-                args.binaries_dir,
+                binaries_dir,
             )
+
+        if not is_source:
+            exts = [".deb"]
         else:
+            exts = [".orig.tar.gz", ".dsc", ".changes", ".debian.tar.xz", ".buildinfo"]
+
+        artifacts_dir = "."
+
+    else:
+        for package in packages:
             build_fedora_package(
-                package, build_deps, run_deps, is_source, args.binaries_dir
+                package,
+                build_deps,
+                run_deps,
+                is_source,
+                distributions,
+                binaries_dir,
             )
 
-os.makedirs(args.output_dir, exist_ok=True)
-if not is_source:
-    if is_ubuntu:
-        exts = [".deb"]
-    else:
-        exts = [".rpm"]
-else:
-    if is_ubuntu:
-        exts = [".orig.tar.gz", ".dsc", ".changes", ".debian.tar.xz", ".buildinfo"]
-    else:
-        exts = [".src.rpm"]
-if is_ubuntu:
-    artifacts_dir = "."
-else:
-    subdir = "SRPMS" if is_source else "RPMS/x86_64"
-    artifacts_dir = f"{home}/rpmbuild/{subdir}"
-for f in os.listdir(artifacts_dir):
-    for ext in exts:
-        if f.endswith(ext):
-            shutil.copy(f"{artifacts_dir}/{f}", os.path.join(args.output_dir, f))
+        if not is_source:
+            exts = [".rpm"]
+        else:
+            exts = [".src.rpm"]
+
+        subdir = "SRPMS" if is_source else "RPMS/x86_64"
+        artifacts_dir = f"{home}/rpmbuild/{subdir}"
+
+    for f in os.listdir(artifacts_dir):
+        for ext in exts:
+            if f.endswith(ext):
+                shutil.copy(f"{artifacts_dir}/{f}", os.path.join(output_dir, f))
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
nitpick, but I believe that it's now unnecessary to have `main()` defined above, that function body could simply be moved here instead

Reply:
We import argument parser defined here in `docker-tezos-packages.py`

Reply:
You can still do that, my point is that we don't import `main()` anywhere, so it doesn't need to be defined separately.

But I don't think this is a major point, so I'll let you decide.